### ISSUE TITLE
Boostrap/Package fails if /usr/bin/protoc is a symlink

### ISSUE THREAD

**Issue by mhthies** [just-buildsystem/justbuild#14]
I'm trying to package *justbuild* for ArchLinux, linking against native shared libraries, provided by ArchLinux packages as far as possible. Thus, I use the bootstrap.py script with the environment variables `PACKAGE=YES LOCALBASE=/usr`.

Another problem with this is that the prebuild `/usr/bin/protoc` executable, provided by [the ArchLinux package *protobuf*](https://archlinux.org/packages/extra/x86_64/protobuf/), is a symbolic link to a more specifically named exectuable file (currently `/usr/bin/protoc-3.21.12.0`). So, even with my workaround from #12, this causes the following problem, as in this case the required file itself is a symlink:

```txt
Running ['/tmp/tmpqz1lqkg3/bootstrap-just', 'analyse', '-C', '/tmp/tmpqz1lqkg3/repo-conf.json', '-D', '{"COMPILER_FAMILY": "clang", "CC": "/usr/bin/clang", "CXX": "/usr/bin/clang++", "AR": "/usr/bin/ar", "ARCH": "x86_64"}', '--dump-graph', '/tmp/tmpqz1lqkg3/graph.json', '--dump-artifacts-to-build', '/tmp/tmpqz1lqkg3/to_build.json', '', ''] in '/tmp/tmpqz1lqkg3/src'
INFO: Requested target is [["@","just","",""],{"AR":"/usr/bin/ar","ARCH":"x86_64","CC":"/usr/bin/clang","COMPILER_FAMILY":"clang","CXX":"/usr/bin/clang++"}]
ERROR: unsupported type for dir entry /usr/include/./pngconf.h
ERROR: unsupported type for dir entry /usr/include/./pybind11
ERROR: unsupported type for dir entry /usr/include/./ncurses.h
[...]
ERROR: unsupported type for dir entry /usr/bin/pgmoil
ERROR: While processing targets:
       While evaluating export target ["@","just","",""]:
       While evaluating install target ["@","just","","installed just"]:
       While evaluating configure target ["@","just","","just"]:
       While evaluating export target ["@","just","","exported-just"]:
       While analysing ["@","rules-just","CC","binary"] target ["@","just","src/buildtool/main","just"]:
       While analysing ["@","rules-just","CC","library"] target ["@","just","src/buildtool/execution_api/local","garbage_collector"]:
       While analysing ["@","rules-just","CC","library"] target ["@","just","src/buildtool/common","bazel_types"]:
       While analysing ["@","rules-just","CC/proto","service library"] target ["#","a0d338d0250537ba57c641b3c933e548bdd24d9b","1181df6b8bbe306c91444bdcd12cda0310104333"]:
       While analysing ["@","rules-protobuf","data","staged"] target ["@","protobuf","","protoc"]:
       While analysing target ["@","protobuf","bin","protoc"] as implicit source target:
       Cannot determine source file "protoc" in directory "bin" of repository "protobuf"
```

To fix this, I patched the `etc/import.prebuild/TARGETS.protobuf` file to search for `bin/protoc-3.21.12.0.` instead of `bin/protoc`. However, this feels more like a workaround than a fix, since I would have to adjust the patch in my packaging recipe whenever the protobuf version is updated.

**Comment by aehlig** [comment]
Thanks for filing this issue. Symbolic links are indeed quite complicated objects. I wrote up my thoughts and suggestions on how to move forward in the presence of symlinks, see #17 . Feel free to comment on that pull request. I assume, this issue, as well as #12  will go away already once we implement the proposed stop-gap measure.

**Comment by aehlig** [comment]
I believe this is fixed in 34ae76b4bfc5d7c153f6c47f5257c2db515e70f1
